Sources/Sampler/py_sampler.hpp

```diff
@@ -152,6 +161,94 @@ void AddSamplerModule(py::module& m) {
   AddExactSampler(subm);
   AddCustomSampler(subm);
   AddMetropolisHastings(subm);
+
+
+  py::class_<MCResult>(subm, "MCResult",
+                       R"EOF(Result of Monte Carlo sampling.)EOF")
+      .def_property_readonly(
+          "samples",
+          [](const MCResult &self) {
+            assert(self.samples.rows() % self.n_chains == 0);
+            return detail::as_readonly(py::array_t<double, py::array::c_style>{
+                {self.samples.rows() / self.n_chains, self.n_chains,
+                 self.samples.cols()},
+                self.samples.data(),
+                py::none()});
+          },
+          py::return_value_policy::reference_internal,
+          R"EOF(Visible configurations `{vᵢ}` visited during sampling.)EOF")
+      .def_property_readonly(
+          "log_values",
+          [](const MCResult &self) {
+            assert(self.log_values.rows() % self.n_chains == 0);
+            return detail::as_readonly(py::array_t<Complex, py::array::c_style>{
+                {self.log_values.rows() / self.n_chains, self.n_chains},
+                self.log_values.data(),
+                py::none()});
+          },
+          py::return_value_policy::reference_internal,
+          R"EOF(An array of `complex128` representing `Ψ(vᵢ)` for all
+                sampled visible configurations `vᵢ`.)EOF");
+
+  subm.def(
+      "compute_samples",
```

Comment:
I think that ```compute_samples``` should just return a rank 2 tensor of visible, and be an easy and plain method defined in ```AbstractSampler``` so that one just does the easy and intuitive thing

```python
samples=sa.compute_samples(n_samples,n_discard)
```

Reply:
What about the log-values?

Reply:
they are not needed, if we want we can implement an internal caching mechanism, that is going to be even faster than just returning log_val

Reply:
That would require changing at least `nk.operator.local_values` as well (and, at least currently, recomputing all the log values there, if I am not mistaken).